src/Functions/DateTimeTransforms.h

```diff
@@ -112,13 +130,17 @@ struct ToStartOfQuarterImpl
 {
     static constexpr auto name = "toStartOfQuarter";
 
-    static inline UInt16 execute(UInt32 t, const DateLUTImpl & time_zone)
+    static inline UInt16 execute(Int64 t, const TimeZoneImpl & time_zone)
     {
         return time_zone.toFirstDayNumOfQuarter(time_zone.toDayNum(t));
     }
-    static inline UInt16 execute(UInt16 d, const DateLUTImpl & time_zone)
+    static inline UInt16 execute(UInt32 t, const TimeZoneImpl & time_zone)
+    {
+        return time_zone.getDefaultLUT().toFirstDayNumOfQuarter(time_zone.toDayNum(t));
+    }
+    static inline UInt16 execute(UInt16 d, const TimeZoneImpl & time_zone)
     {
-        return time_zone.toFirstDayNumOfQuarter(DayNum(d));
+        return time_zone.getDefaultLUT().toFirstDayNumOfQuarter(DayNum(d));
```

Comment:
Wouldn't it better to make wrappers for such functions in Timezone to allow direct call?
```
time_zone.toFirstDayNumOfQuarter(DayNum(d))
```